tests/test_browser.py

```diff
@@ -3538,6 +3538,18 @@ def test_dynamic_link(self):
     # same wasm side module works
     self.btest_exit(self.in_dir('main.c'), args=['-s', 'MAIN_MODULE=2', '-O2', '-s', 'EXPORT_ALL', 'side.wasm'])
 
+  def test_dlopen_async(self):
+    create_file('side.c', 'int foo = 42;\n')
+    self.run_process([EMCC, 'side.c', '-o', 'libside.so', '-s', 'SIDE_MODULE'])
+    self.btest_exit(test_file('other/test_dlopen_async.c'), args=['-s', 'MAIN_MODULE=2'])
+
+  def test_dlopen_blocking(self):
+    create_file('side.c', 'int foo = 42;\n')
+    self.run_process([EMCC, 'side.c', '-o', 'libside.so', '-s', 'SIDE_MODULE'])
+    self.btest_exit(test_file('other/test_dlopen_blocking.c'), args=['-s', 'MAIN_MODULE=2', '-s', 'ASYNCIFY'])
+    # Without the blocking version of dlopen should fail
```

Comment:
can we do a sync version with `PROXY_TO_PTHREAD` yet, or is that future work still?

Reply:
Actually that is not prior work since I landed it in #14528!   So its done.